### PR TITLE
Bring in setup.sh for piksi_tools

### PR DESCRIPTION
**Pull request by mfine**
Bring in supporting scripts from piksi_firmware to setup dependencies for piksi_tools.

/cc @denniszollo @mookerji

### CODE REVIEW

File: setup/ansible/roles/common_osx/tasks/sys_deps.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+                 wsgiref==0.1.2
+                 pyside==1.2.2
+                 pyserial==2.7
+                 pylibftdi==0.14.2
+                 pyparsing==1.5.7
+                 pygments==2.0.2
+                 intelhex==1.5
+                 six
+                 construct
+                 sbp==0.27
```
Comment (mfine) [inline setup/ansible/roles/common_osx/tasks/sys_deps.yml]: ugh - can we replace this with just pip install -r requirements.txt?

File: setup/ansible/roles/common_ubuntu/tasks/sys_deps.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+- name: Handle Ubuntu 13.04 end of life.
+  sudo: True
+  shell: sed -i.bak -r 's/(archive|security).ubuntu.com/old-releases.ubuntu.com/g' /etc/apt/sources.list
+  when: ansible_distribution == 'Ubuntu' and ansible_distribution_version == '13.04'
+
+- name: Install base Ubuntu packages
+  apt: pkg={{ item }} state=installed update_cache=yes
+  sudo: True
+  with_items:
+    - git
```
Comment (mfine) [inline setup/ansible/roles/common_ubuntu/tasks/sys_deps.yml]: Do we need `git` and `build-essential`?

**Comment by mfine** [comment]
This is the mirror of https://github.com/swift-nav/piksi_firmware/pull/396. Towards supporting https://github.com/swift-nav/piksi_tools/issues/40.

File: setup/ansible/roles/common_ubuntu/tasks/sys_deps.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+- name: Update APT repo cache
+  sudo: Yes
+  apt: update_cache=yes
+
+- name: Install python dependencies
+  sudo: True
+  command: >
+             pip install traits traitsui pyserial pylibftdi
+             pyparsing==1.5.7 pygments intelhex six construct sbp==0.29
```
Comment (denniszollo) [inline setup/ansible/roles/common_ubuntu/tasks/sys_deps.yml]: What do we think about getting rid of this line and using requirements.txt for the pip installs?
Similarly, can we use pip to install everything except QT?  In this way manual instructions for other os's will be pretty short.  step 1, get python and Qt. Step to, install pikis_tools,

Comment (mfine) [inline setup/ansible/roles/common_ubuntu/tasks/sys_deps.yml]: We could change this to pip install requirements.txt, but they don't contain all these dependencies right now.

**Comment by cbeighley** [comment]
We should note in the README that there are binaries at downloads.swiftnav.com for those who don't want to install the piksi_tools Piksi Console from source

**Comment by mfine** [comment]
Updated the readme. Should get examples of running the console and bootloader from source in there, too.

File: setup/ansible/roles/common_osx/tasks/sys_deps.yml (hunk 1 above)
Comment (mookerji) [inline setup/ansible/roles/common_osx/tasks/sys_deps.yml]: Does the SBP version need to be updated?

File: setup/ansible/roles/common_osx/tasks/sys_deps.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+- name: Install remaining Python dependencies.
+  sudo: True
+  command: >
+             pip install
+                 traits==4.5.0
+                 traitsui==4.4.0
+                 pyface==4.4.0
+                 wsgiref==0.1.2
+                 pyside==1.2.2
+                 pyserial==2.7
```
Comment (mookerji) [inline setup/ansible/roles/common_osx/tasks/sys_deps.yml]: Do we need pyyaml?

Comment (mfine) [inline setup/ansible/roles/common_osx/tasks/sys_deps.yml]: For the settings.yaml, right?

File: setup/ansible/roles/common_ubuntu/tasks/sys_deps.yml (hunk 2 above)
Comment (cbeighley) [inline setup/ansible/roles/common_ubuntu/tasks/sys_deps.yml]: I think git is required at least for doing a ```git describe``` when building the console binaries.

**Comment by cbeighley** [comment]
@mookerji okay to merge?

File: setup/ansible/roles/common_ubuntu/tasks/sys_deps.yml (hunk 2 above)
Comment (mfine) [inline setup/ansible/roles/common_ubuntu/tasks/sys_deps.yml]: Good point

**Pull request merged**